impl/random.h

```diff
@@ -209,6 +209,51 @@ hydro_random_init(void)
     return ret;
 }
 
+#elif defined(TARGET_LIKE_MBED)
+
+#include "mbedtls/entropy.h"
+#include "mbedtls/ctr_drbg.h"
+
+#if defined(MBEDTLS_ENTROPY_C)
+
+static int
+hydro_random_init(void)
+{
+    mbedtls_entropy_context entropy;
+    mbedtls_entropy_init( &entropy );
+
+    //Pull data directly out of the entropy pool for the state, as it's small enough
+    if(mbedtls_entropy_func( &entropy, (uint8_t *) &hydro_random_context.counter, sizeof(hydro_random_context.counter) ) != 0)
+        return -1;
+
+    // mbedtls_entropy_func can't provide more than MBEDTLS_ENTROPY_BLOCK_SIZE in one go
+    //  This constant depends of mbedTLS configuration (whether the PRNG is backed by SHA256/SHA512 at this time)
+    //  Therefore, if necessary, we get entropy multiple times
+    uint16_t pos = 0;
+    do
+    {
+        const uint8_t dataLeftToConsume = gimli_BLOCKBYTES - pos;
+        const uint8_t currentChunkSize = (dataLeftToConsume > MBEDTLS_ENTROPY_BLOCK_SIZE) ? MBEDTLS_ENTROPY_BLOCK_SIZE : dataLeftToConsume;
+
+        //Forces mbedTLS to fetch fresh entropy
+        mbedtls_entropy_gather( &entropy );
```

Comment:
It isn't (this would be emitted directly by the random source as configured at runtime by the user).
However, I'm not seeing how to act on it. If entropy collection failed, the entropy for the source either wasn't touched or only this source was contaminated. Using this context still provide as much strength as possible at this time.
`mbedtls_entropy_func` also performs a partial reseed so if the error was dramatic, the following call will also fail.